pkg/bff/auth/authtest/server.go

```diff
@@ -0,0 +1,141 @@
+authenticate requests to the BFF server, use this package's token generation methods to
+create a token that will be validated by the authentication middleware. Note that you
+will have to configure the Authenticate middleware to use the correct TLS client.
+*/
+package authtest
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/trisacrypto/directory/pkg/bff/config"
+	"gopkg.in/square/go-jose.v2"
+)
+
+const (
+	KeyID        = "StyqeY8Kl4Eam28KsUs"
+	ClientID     = "a5laOSr0NOX1L53yBaNtumKOoExFxptc"
+	ClientSecret = "me4JZSvBvPSnBaM0h0AoXgXPn1VBiBMz0bL7E/sV1isndP9lZ5ptm5NWA9IkKwEb"
+	Audience     = "http://localhost:4437/"
+	Email        = "leopold.wentzel@gmail.com"
+	UserID       = "test|abcdefg1234567890"
+	OrgID        = "b1b9e9b1-9a44-4317-aefa-473971b4df42"
+	MainNetVASP  = "87d92fd1-53cf-47d8-85b1-048e8a38ced9"
+	TestNetVASP  = "d0082f55-d3ba-4726-a46d-85e3f5a2911f"
+	Scope        = "openid profile email"
+)
+
+// New starts and returns a new Auth0 server using TLS. The caller should call close
+// when finished, to shut it down. The server can also issue tokens for authentication.
+func New() (s *Server, err error) {
+	s = &Server{}
+
+	// Create RSA Private Keys to sign auth tokens with
+	if s.keys, err = rsa.GenerateKey(rand.Reader, 2048); err != nil {
+		return nil, err
+	}
+
+	// Setup routes for the mux
+	s.mux = http.NewServeMux()
+	s.mux.HandleFunc("/.well-known/openid-configuration", s.OpenIDConfiguration)
+	s.mux.HandleFunc("/.well-known/jwks.json", s.JWKS)
+
+	s.srv = httptest.NewTLSServer(s.mux)
+	s.URL, _ = url.Parse(s.srv.URL)
+	return s, nil
+}
+
+// Server wraps an httptest.Server to provide a default handler for auth0 requests.
+type Server struct {
+	srv  *httptest.Server
+	mux  *http.ServeMux
+	URL  *url.URL
+	keys *rsa.PrivateKey
+}
+
+// Config returns an AuthConfig that can be used to setup middleware.
+func (s *Server) Config() config.AuthConfig {
+	return config.AuthConfig{
+		Domain:        s.URL.Host,
+		Audience:      Audience,
+		ProviderCache: 30 * time.Second,
+		ClientID:      ClientID,
+		ClientSecret:  ClientSecret,
+		Testing:       true,
+	}
+}
+
+// Client returns the https configured client that can connect to this server.
+func (s *Server) Client() *http.Client {
+	return s.srv.Client()
+}
+
+// Close the server when you're done with your tests!
+func (s *Server) Close() {
+	s.srv.Close()
+}
+
+// NewToken returns a valid token with the specified permissions.
+func (s *Server) NewToken(permissions ...string) (tks string, err error) {
+	issuer := s.URL.ResolveReference(&url.URL{Path: "/"}).String()
+
+	claims := &Claims{
+		RegisteredClaims: jwt.RegisteredClaims{
+			Issuer:  issuer,
+			Subject: UserID,
+			Audience: jwt.ClaimStrings{
+				issuer,
+				Audience,
+			},
+			IssuedAt:  jwt.NewNumericDate(time.Now()),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(15 * time.Minute)),
```

Comment:
I'm not sure there is any standard for the amount of time claims expire, it mostly depends on the expected use case. Generally these systems hand out two tokens, an access token and a refresh token. The access token ExpiresAt and refresh token NotBefore overlap so that a user can use the refresh token to get a new access token without having to resubmit their username/password. 

The access token expiration acts as an "automatic logout" - so you can combine timings to get a desired application outcome. For example, if you have a high request frequency (maybe something like sensors) you could do an access token expiration of 16 minutes and a refresh token not before of 12 minutes and you're guaranteed that the system will stay online and authenticated but constantly recycling access tokens to prevent replay attacks. 

For web applications, I've generally stuck with 1 hour and 45 minutes so that idle users will get logged out in a reasonable time but if you're actively using the app you'll stay logged in. If the app isn't high security though, then a 24 hour access token would probably work well.